### PR TITLE
New bitsets (experimental)

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

**Comment by Nekotekina** [comment]
This is still not very beautiful, I'd like to improve it before merging.

File: Utilities/bit_set.h [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+	{
+		null = 0, // Empty bitset
+
+		__bitset_set_type = 0 // SFINAE marker
+	};
+
+	static constexpr std::size_t bitmax = sizeof(T) * CHAR_BIT;
+	static constexpr std::size_t bitsize = static_cast<under>(T::__bitset_enum_max);
+
+	static_assert(std::is_enum<T>::value, "bs_t<> error: invalid type (must be enum)");
```
Comment (vlj) [inline Utilities/bit_set.h]: std::is_enum<> returns true for integer type too ?

Comment (Nekotekina) [inline Utilities/bit_set.h]: No, why?

File: rpcs3/Emu/CPU/CPUThread.cpp [hunk 2]
```diff
@@ -104,7 +104,7 @@ bool cpu_thread::check_state()
 			return true;
 		}
 
-		if (!state.test(cpu_state_pause))
+		if (state & cpu_state_pause)
```
Comment (Nekotekina) [inline rpcs3/Emu/CPU/CPUThread.cpp]: Oops.

**Pull request merged**